polygon-daemon/templates/configmap.yaml

```diff
@@ -0,0 +1,134 @@
+    then
+        echo "setting up initial configurations"
+        heimdalld init --home "$HEIMDALLD_HOME"
+        cd $HEIMDALLD_HOME/config
+
+        #config.toml
+        # SEEDS
+        sed -i "s#^seeds.*#seeds = \"${SEEDS}\"#" config.toml
+
+        # heimdall-config.toml
+        # BOR
+        echo "setting up bor rpc"
+        sed -i "s#^bor_rpc_url.*#bor_rpc_url = \"${BOR_RPC_URL}\"#" heimdall-config.toml
+        # ETH
+        echo "setting up eth rpc"
+        sed -i "s#^eth_rpc_url.*#eth_rpc_url = \"${ETH_RPC_URL}\"#" heimdall-config.toml
+
+        echo "removing autogenerated genesis file"
+        rm genesis.json
+
+        echo "downloading correct genesis file"
+        wget -O genesis.json https://raw.githubusercontent.com/maticnetwork/heimdall/master/builder/files/genesis-mainnet-v1.json
+    fi
+
+    # If heimdalld container and we need to bootstrap on first run then download the snapshot
+    if [ ! -n "$REST_SERVER" ] && [ "${BOOTSTRAP}" == 1 ] && [ -n "${SNAPSHOT_URL}" ] && [ ! -f "$HEIMDALLD_HOME/snapshot" ];
+    then
+      echo "downloading snapshot from ${SNAPSHOT_URL}"
+      apk add curl jq zstd aria2
+      mkdir -p ${HEIMDALLD_HOME}/extract && cd ${HEIMDALLD_HOME}/extract
+
+      # download compiled incremental snapshot files list
+      aria2c -x6 -s6 "${SNAPSHOT_URL}"
+
+      # download all incremental files, includes automatic checksum verification per increment
+      aria2c -x6 -s6 -c --auto-file-renaming=false --max-tries=100 --file-allocation=none  -i heimdall-mainnet-incremental-compiled-files.txt
+
+      # marker to indicate that the snapshot process has been completed
+      touch ../snapshot
+    fi
+
+    # Run the correct commands for heimdalld or heimdallr containers
+    #if [ -n "$REST_SERVER" ];
+    #then
+    #  EXEC="heimdalld rest-server --laddr=tcp://0.0.0.0:1317 --home=${HEIMDALLD_HOME} --node=tcp://heimdall:26657"
+    #else
+    #  EXEC="heimdalld start --home=${HEIMDALLD_HOME} --moniker=${MONIKER}  --p2p.laddr=tcp://0.0.0.0:26656 --rpc.laddr=tcp://0.0.0.0:26657"
+    #fi
+
+    exec heimdalld start --home=${HEIMDALLD_HOME} --moniker=${MONIKER}  --p2p.laddr=tcp://0.0.0.0:26656 --rpc.laddr=tcp://0.0.0.0:26657 --rest-server
+    # ${EXEC}
+
+
+  bor-entrypoint.sh: |
+    #!/bin/sh
+
+    # exit script on any error
+    set -e
+
+    # Set Bor Home Directory
+    BOR_HOME=/bor-home
+
+    apk add zstd aria2 curl jq
+
+    # Check for genesis file and download or update it if needed
+    if [ ! -f "${BOR_HOME}/genesis.json" ];
+    then
+        echo "setting up initial configurations"
+        cd ${BOR_HOME}
+        echo "downloading launch genesis file"
+        wget -O genesis.json https://raw.githubusercontent.com/maticnetwork/bor/master/builder/files/genesis-mainnet-v1.json
+    else
+        wget -O genesis.json https://raw.githubusercontent.com/maticnetwork/bor/master/builder/files/genesis-mainnet-v1.json
+    fi
+
+    
```

Comment:
Extra lines here, might be worthwhile to just paste the scripts in a linter/formatter and then copy back into the configmap.